### ISSUE TITLE
CLI --unlock does not work

### ISSUE THREAD

**Issue by sammy007** [ethcore/parity#1798]
`parity --unlock "0x123..." --password /path/to/pass.txt`

I execute `eth_sendTransaction` via JSON RPC:

`Your account is locked. Unlock the account via CLI, personal_unlockAccount or use Trusted Signer`

And of course  I use correct address as rpc param.

If I replace it with `personal_signAndSendTransaction` it works for me, but this op is pretty expensive I guess if you send a lot of continuous payments.

parity` master`.

On 1.2.3 I can't even launch it due to various errors:

```
ERROR: No password given to unlock account 123...456. Pass the password using `--password`.
```

```
ERROR: ~/pass.txt Unable to read password file. Ensure it exists and permissions are correct.
```

Depends on how I specify path. OSX.

**Comment by tomusdrw** [comment]
Thanks for reporting!
Latest master will be fixed with #1802.

But I believe 1.2.3 should work fine, the error you get might indicate that no valid password for this account has been found in file you provided. Can you double check if the password is correct (make sure it uses one password per line and that there is a trailing new line after last password).

Second error will be displayed if you pass file in quotes `--password "~/pass.txt"` - your console will not resolve `~` into home directory.

**Comment by sammy007** [comment]
Password is correct, this is plain text file with just a password followed by newline.

**Comment by tomusdrw** [comment]
Ok, I will double check 1.2.3 then.

**Comment by sammy007** [comment]
@tomusdrw I think some better error message on launch would be cool, like "invalid password" followed by process exit.

**Comment by tomusdrw** [comment]
Yes, I tried to make message more descriptive:
`No password found to unlock account {}. Make sure valid password is present in files passed using `--password`

I've checked 1.2.3 and couldn't reproduce the issue. This works fine for me:
`$ echo "pass" > ./pass && parity --unlock 0x123...456 --password ./pass`

@debris: any ideas? Can it be OSX-specific?

**Comment by ltfschoen** [comment]
I got the same error using Parity v1.8.4-beta `Unable to read password file. Ensure it exists and permissions are correct.` when using a custom `config.toml` file with the following:
```
[account]    
...
password = ["$HOME/Library/Application Support/io.parity.ethereum/testnet/passwords/node.pwds"]
```
I fixed it by replacing `$HOME` with my home directory i.e. `/Users/Ls`

**Comment by chevdor** [comment]
Having the same issue with the current version (master). No matter what I provide as password file, I get `ui Unable to read password file. Ensure it exists and permissions are correct.`

This seems to be an issue parsing what the file should be.

If you call:
`target/release/parity --unlock 0x123... --password /your/path/password.txt --chain kovan ui
`
the message becomes:
`--chain Unable to read password file. Ensure it exists and permissions are correct.
`

=> it considers 
`/your/path/password.txt --chain`
instead of 
`/your/path/password.txt`

**Comment by chevdor** [comment]
workaround:
`$ target/release/parity --chain kovan --unlock 0x13...  --force-ui --password /your/path/password.txt ` but make sure you do NOT provide `--force_ui`after `--password`. `--password ...` should be the last arg.

**Comment by 5chdn** [comment]
This is fixed already, please upgrade your client.

**Comment by chevdor** [comment]
Did you fix it today? I rebuilt today.

**Comment by 5chdn** [comment]
Fixed in #7626 a while ago. See #7428

What's your version string?
